### ISSUE TITLE
Allow screenshot uploaders to delete their own screenshots

### ISSUE THREAD

**Issue by glennlunder** [demozoo/demozoo#99]
@glennlunder Going over the logs for the last night, and I noticed something - http://demozoo.org/productions/534/ - now, I know for a fact that I added those screenshots, likely before we had logging. In this PARTICULAR case, the user tags = Eladamri, who coded the prod, so I expect he had good reason, but the fact that he could just up and delete screenshots like that worries me. Do we have any sort of way to recover screenshots that are deleted, and should we try to tighten that bit up in such a way that for instance only the person who uploaded screenshots and admins can delete?

@gasman Oh. I could have sworn that was always supposed to be admin only - oh well, it is now...  Allowing the uploader to delete their own screenshots is a good idea, but would be non-trivial to implement (in particular, it's going to be tricky to come up with a non-confusing UI that makes it clear why you can delete some screenshots and not others) so I'll ticket that one up for later.

**Comment by asbjornu** [comment]
Yep, I just discovered this. Perhaps allowing users to choose the "key" screenshot for a production would alleviate some of the need for screenshot deletion?

**Comment by glennlunder** [comment]
There's a separate ticket for that - #12
